### ISSUE TITLE
Missing/Incorrect DLL When Publishing net5.0-windows with -p:PublishTrimmed=true

### ISSUE THREAD

**Issue by Vercidium** [dotnet/runtime#55800]
### Description

Publishing applications that use `Microsoft.Management.Infrastructure`, with the command `dotnet publish -c Release -r win7-x64 -f net5.0-windows --self-contained true -p:PublishTrimmed=true` causes the following exception to occur when running the application:

```
System.IO.FileNotFoundException: Could not load file or assembly 'System.Runtime.Handles, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.
File name: 'System.Runtime.Handles, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
   at Microsoft.Management.Infrastructure.CimSession.Create(String computerName, CimSessionOptions sessionOptions)
   at net5trimerror.Helper.GetDedicatedGPUName() in C:\Users\Tesseract\Projects\net5trimerror\Helper.cs:line 16
```

Publishing without `-p:PublishTrimmed=true` allows the application to run fine.

I've attached a small, simple repository here that isolates the issue and shows how to reproduce it:
https://github.com/Vercidium/net5trimerror

### Configuration
.NET Version: `5.0.400-preview.21328.4`
OS Version: `Windows 10.0.19042 Build 19042`
Architecture: `x64, AMD CPU`
I do not know if it is specific to this .NET version

dotnet build uses `Microsoft (R) Build Engine version 16.11.0-preview-21318-04+2eb4b8616 for .NET`
dotnet publish uses `C:\Program Files\dotnet\sdk\5.0.400-preview.21328.4\MSBuild.dll `

**Comment by vitek-karas** [comment]
Trimming in .NET 5 was a Preview feature, so it had lot of "holes" in it. In .NET 6 we improved a lot of these problems. I tried your simple repro with .NET 6 SDK (not targeting .NET 6, just using .NET 6 SDK latest Preview 7 to build the .NET 5 app) and it works.
That said if I retarget the app to .NET 6, there are warnings reported from the trimming process, which indicates that there still might be issues.
